src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -42,6 +42,12 @@ public enum ManagedIdentitySource
         /// <summary>
         /// The source to acquire token for managed identity is Service Fabric.
         /// </summary>
-        ServiceFabric
+        ServiceFabric,
+
+        /// <summary>
+        /// The source to acquire token for managed identity is defaulted to IMDS as no environment variables are found. 
+        /// This is specifically used when the source is not known.
+        /// </summary>
+        DefaultToImds
```

Comment:
We already have an IMDS source, what is the purpose of this one?

Reply:
When no environment variable is found it is not really IMDS but default to IMDS as per the api review https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/12459?_a=files&path=/ManagedIdentity/FacilitateImdsProbing.md. I can leave it as IMDS. But I don't think we should rename the existing enum as it will be a breaking change. So either add a new enum or use exiting one.

Reply:
Gladwin mentioned in the comment below that the pod identity env variable is deprecated. I will use the existing enum